package/MDAnalysis/analysis/contacts.py

```diff
@@ -766,10 +759,157 @@ def plot_qavg(self, filename=None, **kwargs):
+      *u*
+        Universe from which the distance timeseries will be taken
+      *selA*
+        selection string for group A
+      *selB*
+        selection string for group B
+      *radius*
+        distance cutoff for defining reference contacts (Angstroms)
+      *beta*
+        temperature-like, steepness of the shift function for contacts (1/Angstroms)
+      *alpha*
+        tolerance, contact is formed when r < r0*alpha (unitless)
+
+    :Examples:
+
+    1. Protein folding
+
+        ref = Universe("villin.gro")
+        u = Universe("conf_protein.gro", "traj_protein.xtc")
+        Q = calculate_contacts(u, ref, "protein and not name H*", "protein and not name H*")
+
+    2. A pair of helices
+
+        ref = Universe("glycophorin_dimer.pdb")
+        u = Universe("conf_protein.gro", "traj_protein.xtc")
+        Q = calculate_contacts(u, ref, "protein and resid 75-92 and not name H* and segid A", "protein and resid 75-92 and not name H* and segid B")
+
+    :Parameters:
+    For all-atom simulations, radius = 4.5 A and alpha = 1.8 (unitless)
+    For coarse-grained simulations, radius = 6.0 A and alpha = 1.5 (unitless)
+
+    :Reference:
+    Using the definition from Best, Hummer, and Eaton, "Native contacts determine protein folding mechanisms in atomistic simulations" PNAS (2013) 10.1073/pnas.1311599110        
+
+    Eq. (1) of the SI defines the expression for the fraction of native contacts, $Q(X)$:
+    .. math:: $$ Q(X) = \frac{1}{|S|} \sum_{(i,j) \in S} \frac{1}{1 + \exp[\beta(r_{ij}(X) - \lambda r_{ij}^0)]}, $$
+
+    where
+
+        $X$ is a conformation,
+        $r_{ij}(X)$ is the distance between atoms $i$ and $j$ in conformation $X$,
+        $r^0_{ij}$ is the distance from heavy atom i to j in the native state conformation,
+        $S$ is the set of all pairs of heavy atoms $(i,j)$ belonging to residues $\theta_i$ and $\theta_j$ such that $|\theta_i - \theta_j| > 3$ and $r^0_{i,} < 4.5 \unicode{x212B}$,
+        $\beta=5 \unicode{x212B}^{-1}$,
+        $\lambda=1.8$ for all-atom simulations
+
+    """    
+    # reference groups A and B from selection strings
+    refA, refB = ref.select_atoms(selA), ref.select_atoms(selB)
+
+    # 2D float array, reference distances (r0)
+    dref = distance_array(refA.coordinates(), refB.coordinates())
+
+    # 2D bool array, select reference distances that are less than the cutoff radius
+    mask = dref < radius
+    #print("ref has {:d} contacts within {:.2f}".format(mask.sum(), radius))
+
+    # group A and B in a trajectory
+    grA, grB = u.select_atoms(selA), u.select_atoms(selB)
+    results = []
+
+
+    for ts in u.trajectory:
+        d = distance_array(grA.coordinates(), grB.coordinates())
+        r, r0 = d[mask], dref[mask]
+        x = 1/(1 + np.exp(beta*(r - alpha * r0)))
+        print x
+        results.append(( ts.time, x.sum()/mask.sum() ))
+    results = pd.DataFrame(results, columns=["Time (ps)", "Q"])
```

Comment:
With the pandas here, you could hide the import inside a method like..

``` python
@property
def as_dataframe(self):
        try:
            import pandas as pd
        except ImportError:
            raise ImportError("Go get pandas")
        return pd.DataFrame(self.results, columns=[this that etc])
```

So it's completely optional